pyproject.toml

```diff
@@ -27,7 +26,32 @@ ignore = [
     "C408",
     "PERF2",
     "PERF4",
+    "D102",
+    "D105",
+    "D107",
+    "D401",
+    "D203",
+    "D204",
+    "D213",
+    "D215",
+    "D401",
+    "D406",
+    "D407",
+    "D408",
+    "D409",
+    "D413",
+    "D417",
+    "D103"
 ]
+
+[tool.ruff.lint.per-file-ignores]
+".ci*" = ["D"]
+"tests/*" = ["D"]
+"composer/algorithms/*" = ["D"]
+"composer/datasets/*" = ["D"]
+"composer/models/*" = ["D"]
+"composer/trainer/activation_checkpointing.py" = ["D"]
```

Comment:
You can also just add a specific noqa comment at the top of the file to disable it directly in the file.

Reply:
Why ignore algorithms/datasets/models?

Reply:
I just used the same ignoring that pydocstyle used. See the lines removed from the the pre-commit yaml